esp-hal/src/dma/mod.rs

```diff
@@ -2059,9 +2196,13 @@ impl DmaTxBuf {
         let mut buf = Self {
             descriptors,
             buffer,
+            block_size,
         };
         buf.set_length(buf.capacity());
-
+        // no need for block size if the buffer is in DRAM
+        if is_slice_in_dram(buf.buffer) {
+            buf.block_size = None;
+        }
```

Comment:
There should be a check here to make sure block_size is provided when the buffer is in PSRAM

Reply:
I tested small from param with all of the block sizes and they all worked on octal param. I think even none worked - that would have used the default register value. But yea - maybe we should force users to be specific.